arangod/GeneralServer/AuthenticationFeature.cpp

```diff
@@ -101,13 +102,35 @@ void AuthenticationFeature::collectOptions(std::shared_ptr<ProgramOptions> optio
                      new BooleanParameter(&_authenticationUnixSockets));
 #endif
 
+  // Maybe deprecate this option in devel
   options->addOption("--server.jwt-secret",
                      "secret to use when doing jwt authentication",
                      new StringParameter(&_jwtSecretProgramOption));
+
+  options->addOption("--server.jwt-secret-keyfile",
+                     "file containing jwt secret to use when doing jwt authentication.",
+                     new StringParameter(&_jwtSecretKeyfileProgramOption));
 }
 
 void AuthenticationFeature::validateOptions(std::shared_ptr<ProgramOptions>) {
-  if (!_jwtSecretProgramOption.empty()) {
+  if (!_jwtSecretKeyfileProgramOption.empty()) {
+
+    try {
+      // Maybe we want to trim the secret here? Not sure if this is wanted or unwanted
+      // Or base64 encoded?
```

Comment:
I think we should, since whitespace-confusion is very likely at the end of a file. This is probably not so much of a problem in a k8s world, but more so in a starter or manual world.